iothub/service/src/AmqpServiceClient.cs

```diff
@@ -360,6 +364,86 @@ public override async Task SendAsync(string deviceId, string moduleId, Message m
             }
         }
 
+        /// <summary>
+        /// Initiates a new cloud-to-device stream.
+        /// </summary>
+        /// <param name="deviceId">Device Id</param>
+        /// <param name="deviceStreamRequest">Configuration needed for initiating a cloud-to-device stream.</param>
+        /// <param name="cancellationToken">Token used for controlling the termination of the asynchronous call.</param>
+        /// <returns>The result of the cloud-to-device stream request, or null of the request itself could not be completed.</returns>
+        public override Task<DeviceStreamResponse> CreateStreamAsync(string deviceId, DeviceStreamRequest deviceStreamRequest, CancellationToken cancellationToken = default)
+        {
+            return this.CreateStreamAsync(GetDeviceStreamUri(deviceId, deviceStreamRequest.StreamName), deviceStreamRequest, cancellationToken);
+        }
+
+        /// <summary>
+        /// Initiates a new cloud-to-device stream.
+        /// </summary>
+        /// <param name="deviceId">Device Id</param>
+        /// <param name="moduleId">Module ID</param>
+        /// <param name="deviceStreamRequest">Configuration needed for initiating a cloud-to-device stream.</param>
+        /// <param name="cancellationToken">Token used for controlling the termination of the asynchronous call.</param>
+        /// <returns>The result of the cloud-to-device stream request, or null of the request itself could not be completed.</returns>
+        public override Task<DeviceStreamResponse> CreateStreamAsync(string deviceId, string moduleId, DeviceStreamRequest deviceStreamRequest, CancellationToken cancellationToken = default)
+        {
+            return this.CreateStreamAsync(GetModuleStreamUri(deviceId, moduleId, deviceStreamRequest.StreamName), deviceStreamRequest, cancellationToken);
+        }
+
+        internal async Task<DeviceStreamResponse> CreateStreamAsync(
+            Uri uri,
+            DeviceStreamRequest deviceStreamRequest,
+            CancellationToken cancellationToken)
+        {
+            TimeSpan timeout = GetInitiateStreamOperationTimeout(deviceStreamRequest);
```

Comment:
our device streaming code on preview branch probably didn't have logging enabled, we could add it now.

Reply:
I'm not going to invest in this code that much. I just wanted something that would be up-to-date on master that we could refer to when we need to.